include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -523,16 +527,24 @@ class operational_domain_impl
      * border around the domain.
      *
      * @param samples Maximum number of random samples to be taken before flood fill.
+     * @param given_parameter_point Optional parameter point in the parameter space. If it lies within the
+     * operational region, it is used as starting points for flood fill.
      * @return The (partial) operational domain of the layout.
      */
-    [[nodiscard]] operational_domain<parameter_point, operational_status> flood_fill(const std::size_t samples) noexcept
+    [[nodiscard]] operational_domain<parameter_point, operational_status>
+    flood_fill(const std::size_t samples, const std::optional<parameter_point>& given_parameter_point) noexcept
     {
         assert((num_dimensions == 2 || num_dimensions == 3) &&
                "Flood fill is only supported for two and three dimensions");
 
         mockturtle::stopwatch stop{stats.time_total};
 
-        const auto step_point_samples = generate_random_step_points(samples);
+        auto step_point_samples = generate_random_step_points(samples);
+
+        if (given_parameter_point.has_value())
+        {
+            step_point_samples.push_back(to_step_point(given_parameter_point.value()));
+        }
```

Comment:
You're still randomly sampling when a parameter point is provided. Is that the intended behavior? You might land in islands disconnected from the passed parameter point which are then also counted towards the ratio.

Reply:
The number of samples is set to 0 in ``compute_operational_ratio``. But I will change the code to avoid confusion.